### PR TITLE
Add EventListener and EventListenerEventArgs to Mono

### PR DESCRIPTION
**Pull request by qmfrederik**
This PR adds the [EventListener](https://msdn.microsoft.com/en-us/library/system.diagnostics.tracing.eventlistener(v=vs.110).aspx) and [EventWrittenEventArgs](https://msdn.microsoft.com/en-us/library/system.diagnostics.tracing.eventwritteneventargs) classes to Mono. They have been in .NET since 4.5.

Both [EventListener](http://referencesource.microsoft.com/#mscorlib/system/diagnostics/eventing/eventsource.cs,36c1ccbaddc7598f) and [EventWrittenEventArgs](http://referencesource.microsoft.com/#mscorlib/system/diagnostics/eventing/eventsource.cs,9317b6858407017e) are available in the .NET Reference Source.

Unfortunately, it's not possible to just include the files from the reference source, as they take a deep dependency on ETW (see the Design Notes at the top of the file). So the idea was just to stub them.
I may follow up with a PR which adds a very basic implementation of the `EventSource` and `EventListener` classes so that at least, say, logging to a file should work.

**NOTE** Since this PR is adding public classes to the API surface, it is my understanding that the Reference Assemblies should also be updated to include these APIs. I've asked the question over at mono/reference-assemblies/2 but didn't get clear guidance yet on how to get the reference assemblies updated.

Questions, feedback, feel free to let me know.

### CODE REVIEW

File: mcs/class/corlib/System.Diagnostics.Tracing/EventSource.cs [hunk 1]
```diff
@@ -124,6 +126,7 @@ public void Write (string eventName)
 		{
 		}
 
+#if NET_4_6
```
Comment (marek-safar) [inline mcs/class/corlib/System.Diagnostics.Tracing/EventSource.cs]: No need for this, we are now 4.6 only

**Comment by marek-safar** [comment]
Reference assemblies update is tricky. I'll do it once this patch is in

**Comment by qmfrederik** [comment]
Hi @marek-safar 
I removed the #if NET_4_6 statements, let me know there's anything else you

**Comment by marek-safar** [comment]
Please rebase this on top of master

**Pull request merged**